libraries/core/src/main/java/com/paritytrading/philadelphia/FIXSession.java

```diff
@@ -71,7 +71,7 @@
 
     private StringBuilder currentTimestamp;
 
-    FIXSession(Clock clock, SocketChannel channel, FIXConfig config, FIXMessageListener listener,
+    public FIXSession(Clock clock, SocketChannel channel, FIXConfig config, FIXMessageListener listener,
```

Comment:
Perhaps this could have a Javadoc comment like the existing public constructor.

Also, the `Clock` interface should be made public and amended with Javadoc comments. For example:
```java
/**
 * A clock.
 */
public interface Clock {

    /**
     * Get the current time in milliseconds.
     *
     * @return the current time in milliseconds
     * @see System#currentTimeMillis
     */
    long currentTimeMillis();

}
```